staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal/typeconverter.go

```diff
@@ -21,21 +21,52 @@ import (
 
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/kube-openapi/pkg/util/proto"
 	"sigs.k8s.io/structured-merge-diff/typed"
+	"sigs.k8s.io/structured-merge-diff/value"
 	"sigs.k8s.io/yaml"
 )
 
 // TypeConverter allows you to convert from runtime.Object to
 // typed.TypedValue and the other way around.
 type TypeConverter interface {
-	NewTyped(schema.GroupVersionKind) (typed.TypedValue, error)
 	ObjectToTyped(runtime.Object) (typed.TypedValue, error)
 	YAMLToTyped([]byte) (typed.TypedValue, error)
 	TypedToObject(typed.TypedValue) (runtime.Object, error)
 }
 
+// DeducedTypeConverter is a TypeConverter for CRDs that don't have a
+// schema. It does implement the same interface though (and create the
+// same types of objects), so that everything can still work the same.
+// CRDs are merged with all their fields being "atomic" (lists
```

Comment:
This is maybe OK for alpha but I think we need something better for beta and up, can we add a todo or something else to this comment?